shellweb

```diff
@@ -14,23 +15,26 @@ else
 fi
 
 usage() {
-	echo "usage: ${0##*/} [-v] [-a authfile] [-d rootdir] [-p port]" >&2
+	echo "usage: ${0##*/} [-v] [-a authfile] [-d rootdir] [-p port] [-r uri]" >&2
 	exit 1
 }
 
-while getopts "a:d:p:v" OPT; do case $OPT in
-	a)      auth_file=$OPTARG; auth_enabled=true;;
+while getopts "a:d:p:vr:" OPT; do case $OPT in
+	a)      authfile=$OPTARG; auth_enabled=true;;
 	d)      root=$OPTARG;;
 	p)      port=$OPTARG;;
 	v)      verbose=true;;
+	r)	uri=$OPTARG; proxy_mode=true;;
```

Comment:
This isn't a URI, but just a "host" or "host:port" spec. Please rename to smth. like `proxy` or `proxy_spec`, to avoid head scratching in `proxy()` below.